soccer/src/soccer/strategy/agent/position/robot_factory_position.cpp

```diff
@@ -338,4 +351,58 @@ std::string RobotFactoryPosition::get_current_state() {
     return current_position_->get_current_state();
 }
 
+void RobotFactoryPosition::test_play_callback(
+    const rj_msgs::msg::OverridePosition::SharedPtr message) {
+    override_play_position_ =
+        static_cast<Strategy::OverridingPositions>(message->overriding_position);
+}
+
+/**
+ * Checks override_play_position_, which automatically updates when an override is set.
+ * If it is anything but auto, set the current position to that position and return true.
+ */
+bool RobotFactoryPosition::check_for_position_override() {
+    switch (override_play_position_) {
```

Comment:
Not sure about this from a code design perspective. Method name is "check" but it also updates state. Really if we just want to know if a position is being override, that should be a separate method from actually updating that position here